File: deleteColumnsToMakeSorted/deleteColumnsToMakeSoreted.py
from typing import List

class Solution:
    def minDeletionSize(self, strs: List[str]) -> int:
        deleteColumns = 0
        for i in range(len(strs[0])):
            for j in range(len(strs)-1):
                if strs[j][i] > strs[j+1][i]:
                    deleteColumns += 1
                    break
        return deleteColumns

File: deleteColumnsToMakeSorted/deleteColumnsToMakeSoreted.cpp
#include <vector>
#include <string>
//Time Complexity: O(m*n) => matrix traversal since for each word we check each letter
class Solution {
public:
    int minDeletionSize(std::vector<std::string>& strs) {
        int deleteColumns = 0;
        for(int i = 0; i < strs[0].size(); ++i){
            for(int j = 1; j < strs.size(); ++j)
            {
                if(strs[j-1][i] > strs[j][i])
                {
                    ++deleteColumns;
                    break;
                }
            }
        }
        return deleteColumns;
    }
};